samples/command/consumer/src/main.rs

```diff
@@ -71,15 +70,28 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     info!("The Consumer has started.");
 
+    let settings = load_settings();
+
+    let invehicle_digital_twin_url = retrieve_invehicle_digital_twin_url(
+        settings.invehicle_digital_twin_url,
+        settings.chariott_url,
+    )
+    .await?;
+
+    let consumer_authority = settings.consumer_authority;
+
     // Setup the HTTP server.
-    let addr: SocketAddr = CONSUMER_AUTHORITY.parse()?;
+    let addr: SocketAddr = consumer_authority.parse()?;
     let consumer_impl = consumer_impl::ConsumerImpl::default();
     let server_future =
         Server::builder().add_service(DigitalTwinConsumerServer::new(consumer_impl)).serve(addr);
-    info!("The HTTP server is listening on address '{CONSUMER_AUTHORITY}'");
+    info!("The HTTP server is listening on address '{consumer_authority}'");
+
+    // This is a workaround: see https://stackoverflow.com/questions/23975391/how-to-convert-a-string-into-a-static-str
+    let static_url_str = Box::leak(invehicle_digital_twin_url.into_boxed_str());
```

Comment:
Try removing `static_url_str` variable, and just pass `invehicle_digital_twin_url` as a reference directly into your `discover_digital_twin_provider_using_ibeji(&invehicle_digital_twin_url, ...)`. 

Inside your `discover_digital_twin_provider_using_ibeji` function, the `DigitalTwinClient::connect` function can accept Strings instead of &'static str.


```
let mut client = DigitalTwinClient::connect(invehicle_digitial_twin_service_url.to_string())
        .await
        .map_err(|error| format!("{error}"))?;
```

Reply:
Great suggestion!  Fixed.